libs/api/domains/auth/src/lib/services/types.ts

```diff
@@ -0,0 +1,88 @@
+import { User } from '@island.is/auth-nest-tools'
+import {
+  CreateDelegationInput,
+  DelegationInput,
+  DelegationsInput,
+  DeleteDelegationInput,
+  PatchDelegationInput,
+  UpdateDelegationInput,
+} from '../dto'
+import { DelegationByOtherUserInput } from '../dto/delegationByOtherUser.input'
+import { ApiScope } from '../models'
+import { ScopeTreeNode } from '../models/scopeTreeNode.model'
+import { ApiScopesInput } from '../dto/apiScopes.input'
+
+export interface MeDelegationsServiceI {
+  getDelegations(user: User, input: DelegationsInput): Promise<DelegationDTO[]>
+
+  getDelegationById(
+    user: User,
+    input: DelegationInput,
+  ): Promise<DelegationDTO | null>
+
+  getDelegationByOtherUser(
+    user: User,
+    input: DelegationByOtherUserInput,
+  ): Promise<DelegationDTO | null>
+
+  createOrUpdateDelegation(
+    user: User,
+    input: CreateDelegationInput,
+  ): Promise<DelegationDTO>
+
+  deleteDelegation(user: User, input: DeleteDelegationInput): Promise<boolean>
+
+  updateDelegation(
+    user: User,
+    input: UpdateDelegationInput,
+  ): Promise<DelegationDTO>
+
+  patchDelegation(
+    user: User,
+    input: PatchDelegationInput,
+  ): Promise<DelegationDTO>
+}
+
+export interface ApiScopeServiceI {
```

Comment:
Would prefer `ApiScopeServiceInterface` over `I` postfix. And maybe as a general naming convention when we are actually using interface as a service contract instead of data modeling.

Reply:
Agreed!